### PR TITLE
Add version skew policy document

### LINKED ISSUE #4916: Please describe version skew policy between Gardener components
**What would you like to be added**:
Documentation, much like https://kubernetes.io/releases/version-skew-policy, that describes the allowed version skew between Gardener components, e.g. `gardener-apiserver`, `gardener-controller-manager`, `gardener-scheduler`, `gardenlet`s in self-managed seeds, etc.

**Why is this needed**:
For operators to understand how/when they need to update Gardener components, e.g. allowed skew between manually managed `gardenlet`s and the Gardener control plane.

cc @rfranzke

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area documentation open-source
/kind enhancement

**What this PR does / why we need it**:
This PR adds a version skew policy document.

Practically, we always followed and proclaimed this policy, however, it was not documented yet.
Also, since there are two PRs which depend on v1.37 to be deployed (#5121 and #5128), let's now make this policy official.

**Which issue(s) this PR fixes**:
Fixes #4916

**Release note**:

```noteworthy operator
An official version skew policy document was added. You can take a look [here](https://github.com/gardener/gardener/blob/master/docs/deployment/version_skew_policy.md).
```

### CODE REVIEW

File: docs/deployment/version_skew_policy.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+This document describes the maximum version skew supported between various Gardener components.
+
+## Supported Gardener Versions
+
+Gardener versions are expressed as `x.y.z`, where `x` is the major version, `y` is the minor version, and `z` is the patch version, following Semantic Versioning terminology.
+
+The Gardener project maintains release branches for the most recent three minor releases.
+
+Applicable fixes, including security fixes, may be backported to those three release branches, depending on severity and feasibility.
+Patch releases are cut from those branches at a regular cadence, plus additional urgent releases, when required.
```
Comment (acumino) [inline docs/deployment/version_skew_policy.md]:
```suggestion
Applicable fixes, including security fixes, maybe backported to those three release branches, depending on severity and feasibility.
Patch releases are cut from those branches at a regular cadence, plus additional urgent releases when required.
```

File: docs/deployment/version_skew_policy.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+The supported version skew between components has implications on the order in which components must be upgraded.
+This section describes the order in which components must be upgraded to transition an existing Gardener installation from version **1.37** to version **1.38**.
+
+#### gardener-apiserver
+
+Pre-requisites:
+
+- In a single-instance setup, the existing `gardener-apiserver` instance is **1.37**
+- In a multi-instance setup, all `gardener-apiserver` instances are at **1.37** or **1.38** (this ensures maximum skew of 1 minor version between the oldest and newest `gardener-apiserver` instance)
+- The `gardener-controller-manager`, `gardener-scheduler`, `gardener-admission-controller`, and `gardenlet` instances that communicate with this `gardener-apiserver` are at version **1.37** (this ensures they are not newer than the existing API server version, and are within 1 minor version of the new API server version)
```
Comment (acumino) [inline docs/deployment/version_skew_policy.md]:
```suggestion
- The `gardener-controller-manager`, `gardener-scheduler`, `gardener-admission-controller`, and `gardenlet` instances that communicate with this `gardener-apiserver` are at version **1.37** (this ensures they are not newer than the existing API server version and are within 1 minor version of the new API server version)
```

File: docs/deployment/version_skew_policy.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+Applicable fixes, including security fixes, may be backported to those three release branches, depending on severity and feasibility.
+Patch releases are cut from those branches at a regular cadence, plus additional urgent releases, when required.
+
+For more information, see [this document](../development/process.md#releases).
+
+### Supported Version Skew
+
+Technically, we follow the same [policy](https://kubernetes.io/releases/version-skew-policy/) as the Kubernetes project.
+However, given that our release cadence is much more frequent compared to Kubernetes (every `14d` vs. every `120d`), in many cases it is possible to skip a version.
+Still, to be on the safe side, it is highly recommended following the described policy.
```
Comment (acumino) [inline docs/deployment/version_skew_policy.md]:
```suggestion
However, given that our release cadence is much more frequent compared to Kubernetes (every `14d` vs. every `120d`), in many cases, it is possible to skip a version.
Still, to be on the safe side, it is highly recommended to follow the described policy.
```

**Review by acumino — CHANGES_REQUESTED** [review]

File: docs/deployment/version_skew_policy.md (hunk 1 above)
Comment (rfranzke) [reply docs/deployment/version_skew_policy.md]: While I removed the `,`, the `may be` is intended (`maybe` means something different).

File: docs/deployment/version_skew_policy.md (hunk 3 above)
Comment (rfranzke) [reply docs/deployment/version_skew_policy.md]: While I switched `following` to `to follow`, the `,` seems wrong, so I haven't added it.

**Comment by rfranzke** [comment]
Thanks @acumino, please check again :)

**Review by timuthy — DISMISSED** [review]
/lgtm
Nice document 🙂

**Review by acumino — DISMISSED** [review]

File: docs/deployment/version_skew_policy.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+# Version Skew Policy
```
Comment (ialidzhikov) [inline docs/deployment/version_skew_policy.md:1]: Does it make sense to drop now the short section that was introduced with https://github.com/gardener/gardener/pull/4917 as this document provides much more details on the topic?

Comment (rfranzke) [reply docs/deployment/version_skew_policy.md:1]: Yep, I've dropped it. thanks!

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**